[FLINK-18152][Depl] Fail fast in JM scripts if memory configuration failed

azagrebin opened the pull request:
Add check whether the memory configuration failed in JM bash scripts:
```
if [[ $? -ne 0 ]]; then
  echo "[ERROR] Could not get JVM parameters and dynamic configurations properly."
  exit 1
fi
```

tillrohrmann commented on flink-dist/src/main/flink-bin/bin/jobmanager.sh:
```diff
@@ -40,6 +40,10 @@ if [[ $STARTSTOP == "start" ]] || [[ $STARTSTOP == "start-foreground" ]]; then
     # Add JobManager-specific JVM options
     export FLINK_ENV_JAVA_OPTS="${FLINK_ENV_JAVA_OPTS} ${FLINK_ENV_JAVA_OPTS_JM}"
     parseJmJvmArgsAndExportLogs "${ARGS[@]}"
+    if [[ $? -ne 0 ]]; then
+      echo "[ERROR] Could not get JVM parameters and dynamic configurations properly."
+      exit 1
+    fi
```
Isn't this a bit redundant given that we do the check in `parseJmJvmArgsAndExportLogs`?

tillrohrmann commented on flink-dist/src/main/flink-bin/bin/standalone-job.sh:
```diff
@@ -40,6 +40,10 @@ if [[ $STARTSTOP == "start" ]] || [[ $STARTSTOP == "start-foreground" ]]; then
     # Add cluster entry point specific JVM options
     export FLINK_ENV_JAVA_OPTS="${FLINK_ENV_JAVA_OPTS} ${FLINK_ENV_JAVA_OPTS_JM}"
     parseJmJvmArgsAndExportLogs "${ARGS[@]}"
+    if [[ $? -ne 0 ]]; then
+      echo "[ERROR] Could not get JVM parameters and dynamic configurations properly."
+      exit 1
+    fi
```
Same here.

tillrohrmann commented on flink-dist/src/main/flink-bin/mesos-bin/mesos-jobmanager.sh:
```diff
@@ -27,6 +27,10 @@ bin=$(cd "${bin}" || exit; pwd)
 . "${bin}"/config.sh
 
 parseJmJvmArgsAndExportLogs "${ARGS[@]}"
+if [[ $? -ne 0 ]]; then
+  echo "[ERROR] Could not get JVM parameters and dynamic configurations properly."
+  exit 1
+fi
```
And here.

azagrebin commented:
Thanks for the review @tillrohrmann 
This is true, the other checks are redundant as the exit code is lost only when the result of call is assigned to a variable. I addressed the comments.
